### DISCUSSION TITLE
Using more than 1 EBSD file to obtain one ODF

### DISCUSSION THREAD

**Discussion by bpuentes94** [mtex-toolbox/mtex#1121] [Ask Anything]
Hi there,

I am using Mtex to process EBSD data and plot it into an ODF (Orientation Distribution Function).
One issue I am facing is that for fully recrystallized samples, the EBSD data is not large enough (big grains = less number of grains in one scan) to produce a proper ODF. I would like to increase the amount of data by adding several EBSD scans into one ODF

Is it possible to do this?

I tried to use the import wizard tool and several files at once, but this didn't work.

I appreciate any advice, thanks for your time.

**Comment by zmichels** [comment]
Hi,

Likely someone will chime in with a more elegant solution than mine, but this is what I would do…

1. Compute grains for one sample.
2. Make a variable for the meanOrientation of the grains in that sample.
3. Save that variable in a .MAT file, or export the Euler angles to some other format (.TXT,.CSV,.XLS).
4. Repeat for other samples.
5. After you think you’ve processed enough grains… import each previous dataset and aggregate them into a single variable that contains all the orientations.
6. Compute an ODF from that aggregated set of orientation.

Hope that helps.

Cheers,
Zach

**Reply by kilir** [reply]
Hi,
you can also take orientations e.g. all EBSD points or grain meanOrientation assign them a variable name you don't overwrite `oris1 = ebsd('xy').orientations` load another dataset and `oris2 = ebsd('xy').orientations` and so on and concatenate those `alloris = [oris1 oris2 ...]` and compute the odf from those.

Cheers,
Rüdiger
